packages/block-library/src/navigation/edit/menu-inspector-controls.js

```diff
@@ -109,23 +80,45 @@ const MainContent = ( {
 				'You have not yet created any menus. Displaying a list of your Pages'
 		  );
 
-	const renderLinkUI = ( block ) => {
+	const renderLinkUI = (
+		currentBlock,
+		lastInsertedBlock,
+		setLastInsertedBlock
+	) => {
+		const blockHasExistingLinkValue = !! lastInsertedBlock?.url;
+		const blockSupportsLinkUI = BLOCKS_WITH_LINK_UI_SUPPORT?.includes(
+			lastInsertedBlock?.name
+		);
+		const currentBlockWasJustInserted =
+			lastInsertedBlock?.clientId === currentBlock.clientId;
+
+		const shouldShowLinkUIForBlock =
+			blockSupportsLinkUI &&
+			! blockHasExistingLinkValue && // don't re-show the Link UI if the block already has a link value.
```

Comment:
Is this check necessary? If it only appears after a user has inserted the block, it should never have a link, right? If so, I think we can remove all the checks for if there's a link or not.

Reply:
Yeh you could be right there. Have you got time to test that and commit as it's nearly my EOD?